### ISSUE TITLE
A missing comma results in merging objects

### ISSUE THREAD

**Issue by pokle** [google/jsonnet#537]
I hope this is due to my misunderstanding of how jsonnet works. If I leave out a comma between objects, jsonnet merges the two objects. This is a very easy mistake to make.

For example:
```shell
$ echo '[{a:1}{b:2}]' | jsonnet -
[
   {
      "a": 1,
      "b": 2
   }
]
```

**Comment by jordansissel** [comment]
I ran into this tonight. The behavior is quite easy to mistakenly cause and debugging it is tricky. Given this kind of easy mistake, I propose removing this feature (having `obj {...}` be shorthand for `obj + {...}`). 

The shorthand isn't really much shorthand (saves 1 byte?) and is easy to overlook and harder to debug.

I'd guess there's probably historical reasons for why this exists, but as there is no linter at this time, and linting is usually optional, would you accept a patch removing this particular grammar (`expr { objinside }`)  in the effort to improve user experience?

**Comment by sbarzowski** [comment]
I fully agree, but we definitely cannot just remove it - that would break many existing users. It is something which could perhaps be switched off with a pragma one day, but I wouldn't hold my breath.

**Comment by sparkprime** [comment]
Technically this exists with functions too:
`[f(1 + 2)]` vs `[f, (1 + 2)]`

**Comment by sparkprime** [comment]
The linter could require the + in contexts where a comma would also be allowed.  That's quite a lot of contexts though I think.

**Comment by tsnobip** [comment]
This is indeed a very unfortunate syntactic choice, I got bitten by it and it's quite hard to debug. 

Adding it to the linter and eventually sunsetting it would be a good idea.
